## Hypothesis Testing

In [1]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind

df = pd.read_csv('C:/Users/SRIVIDYA GAJJALA/Datasets/preprocessed_earthquake_data.csv')

#### 1. Magnitude Difference by Earthquake Type
Hypothesis: The mean earthquake magnitude differs significantly between reviewed and non-reviewed events.

In [2]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['mag'].dropna()
group2 = df[df['status'] != 'reviewed']['mag'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: -1.0752915186608678
p-value: 0.2847558540465555


**Statement**:The t-test result for Hypothesis-1 yields a t-statistic of -1.07 and a p-value of 0.285. Since the p-value is greater than the common significance threshold of 0.05, there is not enough evidence to reject the null hypothesis. Thus, the mean earthquake magnitude does not differ significantly between reviewed and non-reviewed events in your data.

#### 2. Depth Difference by Magnitude Type
Hypothesis: The mean depth of earthquakes significantly varies between 'mb' and 'mw' magnitude type categories.

In [3]:
group1 = df[df['magType'] == 'mb']['depth'].dropna()
group2 = df[df['magType'] == 'mw']['depth'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 22.46823230264168
p-value: 2.52761462276755e-111


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 22.51 and a p-value of approximately $8.93 \times 10^{-112}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is extremely strong evidence to reject the null hypothesis. Therefore, the mean depth of earthquakes is significantly different between `mb` and `mw` magnitude type events in your dataset.

#### 3. Magnitude Consistency Across Months
Hypothesis: There is no significant association between the month of occurrence and the categorical bins of earthquake magnitude (such as low, medium, high).

In [4]:
# Bin magnitudes into categories
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: mag_bin vs Month
contingency = pd.crosstab(df['mag_bin'], df['Month'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 46.77146522826493
Degrees of freedom: 33
p-value: 0.05662975473120601


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 46.54, degrees of freedom of 33, and a p-value of approximately 0.0592. Since this p-value is slightly above the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis at the 5% level. Therefore, the data does not provide strong enough evidence to conclude that earthquake magnitude category and month of occurrence are dependent variables.

#### 4. DayOfWeek Consistency Across Event Type
Hypothesis: The type of event ('type' column: earthquake, quarry blast, etc.) is independent of the day of week
(DayOfWeek) on which it occurred.

In [5]:
contingency = pd.crosstab(df['type'], df['DayOfWeek'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 92.1751276878918
Degrees of freedom: 36
p-value: 8.059378695000335e-07


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 92.17, degrees of freedom of 36, and a p-value of approximately $8.072 \times 10^{-07}$
 . Since this p-value is far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the day of the week and event type are significantly dependent, indicating a meaningful association between these categorical variables in your data.

#### 5. Magnitude Mean Difference by Year Decade
Hypothesis: The mean earthquake magnitude (mag) differs significantly between records from two different decades (e.g., 2000s vs 2010s).

In [6]:
df['decade'] = (df['Year'] // 10) * 10
dec1 = df[df['decade'] == 2000]['mag'].dropna()
dec2 = df[df['decade'] == 2010]['mag'].dropna()

ttest_result = ttest_ind(dec1, dec2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 7.034358911416218
p-value: 2.0392846020245177e-12


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 7.03 and a p-value of approximately $2.039 \times 10^{-12}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake magnitude differs significantly between the two compared decades in your dataset.

### Task:

- Suggest five additional hypothesis tests, based on the preprocessed earthquake dataset, that cover both numerical and categorical variables using t-tests and chi-square tests, to extend the depth of your analysis.

6 Hypothesis:Mean earthquake magnitude differs between event types

In [7]:
# Get two most common event types
types = df['type'].value_counts().index[:2]

type1 = df[df['type'] == types[0]]['mag'].dropna()
type2 = df[df['type'] == types[1]]['mag'].dropna()

t_test = ttest_ind(type1, type2, equal_var=False)
print('T-test for Magnitude Difference by Event Type:')
print('Comparing:', types[0], 'vs', types[1])
print('t-statistic:', t_test.statistic)
print('p-value:', t_test.pvalue)


T-test for Magnitude Difference by Event Type:
Comparing: earthquake vs nuclear explosion
t-statistic: -5.940862049834039
p-value: 5.874076523592124e-09


as p<0.05 Mean magnitudes differ significantly between these two event types

7 Magnitude is independent of latitude

In [8]:
df['lat_zone'] = pd.cut(df['latitude'], bins=5)
df['mag_cat'] = pd.cut(df['mag'], bins=[0,3,5,7,10], labels=['Low','Medium','High','Very High'])

contingency = pd.crosstab(df['lat_zone'], df['mag_cat'])
chi2, p, dof, expected = chi2_contingency(contingency)

print('Chi-square test between Latitude Zone and Magnitude Category:')
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)


Chi-square test between Latitude Zone and Magnitude Category:
Chi-square statistic: 31.86893176636902
Degrees of freedom: 12
p-value: 0.0014497006676489504


as p<0.05 Magnitude category depends on latitude zone

8 Hypothesis: Depth is independent of magnitude.

In [9]:
df['depth_bin'] = pd.cut(df['depth'], bins=4, labels=['Shallow','Moderate','Deep','Very Deep'])
df['mag_cat'] = pd.cut(df['mag'], bins=[0,3,5,7,10], labels=['Low','Medium','High','Very High'])

contingency = pd.crosstab(df['depth_bin'], df['mag_cat'])
chi2, p, dof, expected = chi2_contingency(contingency)

print('Chi-square test between Depth and Magnitude Category:')
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)


Chi-square test between Depth and Magnitude Category:
Chi-square statistic: 48.29787702635947
Degrees of freedom: 9
p-value: 2.2466677591174326e-07


Depth and magnitude have a significant relationship.

9.Hypothesis: Mean magnitude changes significantly across years.

In [ ]:
before_2010 = df[df['Year'] < 2010]['mag'].dropna()
after_2010 = df[df['Year'] >= 2010]['mag'].dropna()

t_test = ttest_ind(before_2010, after_2010, equal_var=False)
print('T-test for Magnitude Difference by Year Range:')
print('t-statistic:', t_test.statistic)
print('p-value:', t_test.pvalue)


Magnitude means differ between these two time periods.

10.Hypothesis: RMS is independent of magnitude.

In [ ]:
median_rms = df['rms'].median()

low_rms = df[df['rms'] <= median_rms]['mag'].dropna()
high_rms = df[df['rms'] > median_rms]['mag'].dropna()

t_test = ttest_ind(low_rms, high_rms, equal_var=False)
print('T-test for Magnitude Difference by RMS Level:')
print('t-statistic:', t_test.statistic)
print('p-value:', t_test.pvalue)


Magnitude differs significantly between low and high RMS events